### PR TITLE
PCHR-3167: Improve UX of deduction fields for Leave in Hours

### PR DESCRIPTION
**Pull request by igorpavlov**
## Overview

It might be unclear for end users how deductions work and are they actually attached to From and To days only in case of a multiple day request. This PR attempts to eliminate potential confusion with UI. It attempts to explicitly show that a deduction is attached to a specific date, shows default suggested deduction and optionally allows to edit it.

Some changes are also introduced in https://github.com/compucorp/civihr-employee-portal-theme/pull/313

#### Desktop

2admin

#### Desktop

2ssp

#### Mobile

2sspmob

## Technical Details

```html
<crm-loading show="!detailsTab.uiOptions.times.from.loading"
  ng-if="detailsTab.selectedAbsenceType.calculation_unit_name === 'hours' && !detailsTab.isLeaveType('toil') && detailsTab.request.from_date && detailsTab.uiOptions.times.from.maxAmount > 0 && !__isNotEditable"
  class="chr_leave-request-modal__small_spinner">
  <div class="row form-group"
    ng-switch="detailsTab.uiOptions.times.from.amountExpanded">
    <div ng-switch-when="false"
      class="chr_leave-request-modal__small-text chr_leave-request-modal__custom-deduction-toggling-row">
      <!-- Different setups for mobile/desktop -->
      <div class="col-sm-8 col-sm-offset-4 col-xs-12">
        <!-- Small label for information -->
        <strong>{{detailsTab.uiOptions.times.from.amount | timeUnitApplier : detailsTab.selectedAbsenceType.calculation_unit_name}}</strong>
        for {{detailsTab.uiOptions.fromDate | formatDate}}
        <a class="pointer"
          ng-click="detailsTab.uiOptions.times.from.amountExpanded = true">
          <!-- Allowing to customise deduction -->
          or set custom deduction</a>
      </div>
    </div>
    <div ng-switch-when="true">
      <div class="col-sm-6 chr_leave-request-modal__small-text chr_leave-request-modal__custom-deduction-label">
        <span>Customise deduction for </span>
        <strong>{{detailsTab.uiOptions.fromDate | formatDate}}</strong>
        <br class="hidden-xs"/>
        <span class="text-muted">(ex. exclude lunch time)</span>
      </div>
      <div class="col-sm-6">
        <a class="btn btn-sm btn-link text-muted pull-right"
          ng-click="detailsTab.uiOptions.times.from.amountExpanded = false">
          <i class="fa fa-times"></i>
        </a>
        <!-- The same time amount picker that was used previously -->
        <time-amount-picker
          time-amount-picker-value="detailsTab.uiOptions.times.from.amount"
          time-amount-picker-min-amount="0"
          time-amount-picker-max-amount="detailsTab.uiOptions.times.from.maxAmount"
          time-amount-picker-interval="15"
        ></time-amount-picker>
      </div>
    </div>
  </div>
</crm-loading>
```

## Comments

Deductions fields are not visible at all when you are in the "view" mode since these deductions are covered in the balance breakdown.

The "disabled" attribute is not needed anymore for deductions.

### New Backstop scenarios

civihr_leave_and_absences_my_report_leave_request_modal_create_show_deductions_fields_0_document_0_desktop

civihr_leave_and_absences_my_report_leave_request_modal_create_show_deductions_fields_expanded_0_document_0_desktop

----

✅ Manual Tests - passed
✅ Jasmine Tests - amended and passed
✅ JS distributive files - included
✅ CSS distributive files - included
✅ Backstop tests - partially checked, passed
⏹ PHP Unit Tests - not needed

### CODE REVIEW

File: uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/components/leave-request-popup-details-tab.component.js [hunk 1]
```diff
@@ -98,17 +98,17 @@ define([
           max: '00:00',
           amount: 0,
           maxAmount: 0,
-          disabled: true,
-          loading: false
+          loading: false,
+          amountExpanded: false
```
Comment (reneolivo) [inline uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/components/leave-request-popup-details-tab.component.js]: Lets try to sort these properties in alphabetical order.

File: uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/components/leave-request-popup-details-tab.component.js [hunk 2]
```diff
@@ -98,17 +98,17 @@ define([
...
         to: {
           time: '',
           min: '00:00',
           max: '00:00',
           amount: 0,
           maxAmount: 0,
-          disabled: true,
-          loading: false
+          loading: false,
+          amountExpanded: false
```
Comment (reneolivo) [inline uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/components/leave-request-popup-details-tab.component.js]: these as well.

File: uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/components/leave-request-popup-details-tab.component.js (hunk 1 above)
Comment (igorpavlov) [reply uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/components/leave-request-popup-details-tab.component.js]: Done.

File: uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/components/leave-request-popup-details-tab.component.js (hunk 2 above)
Comment (igorpavlov) [reply uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/components/leave-request-popup-details-tab.component.js]: Done.

**Pull request merged**